Marlin/src/pins/stm32f1/pins_MKS_ROBIN.h

```diff
@@ -41,24 +43,22 @@
 #define DISABLE_JTAG
 
 //
-// Enable SD EEPROM to prevent infinite boot loop
+// EEPROM
 //
-#ifdef ARDUINO_ARCH_STM32
-  #define FLASH_EEPROM_EMULATION
-  #define EEPROM_PAGE_SIZE     (0x800U) // 2KB
-  #define EEPROM_START_ADDRESS (0x8000000UL + (STM32_FLASH_SIZE) * 1024UL - (EEPROM_PAGE_SIZE) * 2UL)
-  #define MARLIN_EEPROM_SIZE (EEPROM_PAGE_SIZE)
-#else
-  #define SDCARD_EEPROM_EMULATION
+#if NO_EEPROM_SELECTED
+  #define IIC_EEPROM_SDA                    PB7
+  #define IIC_EEPROM_SCL                    PB6
+  #define MARLIN_EEPROM_SIZE                0x800  // 2KB
+  #undef NO_EEPROM_SELECTED
 #endif
 
 //
 // Servos
 //
-#define SERVO0_PIN                          PC3   // XS1 - 5
-#define SERVO1_PIN                          PA1   // XS1 - 6
-#define SERVO2_PIN                          PF9   // XS2 - 5
-#define SERVO3_PIN                          PF8   // XS2 - 6
+#define SERVO0_PIN                          PA1   // XS1 - 6 (BLTOUCH)
+#define SERVO1_PIN                          PC3   // XS1 - 5
+#define SERVO2_PIN                          PF8   // XS2 - 6
+#define SERVO3_PIN                          PF9   // XS2 - 5
```

Comment:
Is rearranging these likely to break existing users?

Did they perhaps move connectors around between board revisions to provide a dedicated BLTouch connector or something?

Reply:
I only rearranged SERVO PINS like this because it seemed like PA1 is the only hardware PWM capable pin according to the datasheet so i decided to use it for BLTOUCH as it should apparently be the preferred servo pin.

Reply:
Currently hardware PWM is never used for Servo output in our STM32 HALs, including for a BLTouch.
That doesn't mean it _couldn't_ be used, it just is not today.

Timer conflicts and shortages have been a battle on some of these devices, so on many boards it isn't possible to find pins capable of hardware PWM at all.

Does this board even have pins marked Servo, or are these just on the AUX connector?
It seems bizarre to declare four servo pins at all if they don't actually exist on the board. I can see defining one specifically for a BLTouch, but even that is questionable when the board doesn't have a labeled position for it.

Reply:
Oh, I see the pins this refers to now.

Reply:
In fact, these pins are neither servo nor BLTouch marked. Some Robin variants (not covered by this file) have dedicated BLTouch or servo pins but it is not the case for the original Robin board. There are only some expansion headers (simple 2.54mm dual rank pin headers) with some weird pinouts. I didn't know that Marlin cannot make use of hardware PWM, as it _may_ be implemented one day it should still be a good idea to keep PA1 as the preferred servo pin, what do you think ?

Reply:
I _personally_ prefer to leave them the same, so that if anybody is already using Servo0 it will not stop working.

Reply:
I understand, and i think that if there is a way to specify the servo pin used for BLTouch it is the right way to go. If not i think that maybe the feedback of more Robin users like @mattdog01 would make us decide.

Reply:
When a servo probe or BLTouch is in use, the `Z_PROBE_SERVO_NR` defines which servo port it uses.